astropy/utils/decorators.py

```diff
@@ -751,3 +752,230 @@ def _get_function_args(func, exclude_args=()):
                 all_args[arg_type] = None
 
     return all_args
+
+
+def format_doc(docstring, *args, **kwargs):
+    """
+    Replaces the docstring of the decorated object and then formats it.
+
+    This decorator keeps the objects signature and only alters the saved
+    ``__doc__`` property (so it cannot be used as class decorator in python2).
+    The formatting works like :meth:`str.format` and if the decorated object
+    already has a docstring this docstring can be included in the new
+    documentation if you use the ``{__doc__}`` placeholder.
+    It's primary use is if multiple functions have the same or only
+    a slightly different *long* docstring and you want to DRY
+    (https://de.wikipedia.org/wiki/Don%E2%80%99t_repeat_yourself).
+
+    Parameters
+    ----------
+    docstring : str or object
+        The docstring that will replace the docstring of the decorated
+        object. If it is an object like a function or class it will
+        take the docstring of this object. If it is a string it will use the
+        string itself. One special case is if the string is ``'__doc__'`` then
+        it will use the decorated functions docstring and formats it.
```

Comment:
I can't decide how I feel about this "special case".  Part of me wants to just say, just use ``{{__doc__}}``.  But this is also convenient.  Maybe I'd prefer if the "special case" were just `None`.